File: matcher/src/models/product.py
from pydantic import BaseModel


class SourceProduct(BaseModel):
    reference: str
    name: str
    brand: str | None = None
    ean: str | None = None
    category: str | None = None
    price: float | None = None
    attributes: dict = {}


class TargetProduct(BaseModel):
    reference: str
    name: str
    brand: str | None = None
    ean: str | None = None
    retailer: str | None = None
    url: str | None = None
    price: float | None = None
    category: str | None = None
    attributes: dict = {}


class CompetitorMatch(BaseModel):
    reference: str
    competitor_retailer: str | None = None
    competitor_product_name: str | None = None
    competitor_url: str | None = None
    competitor_price: float | None = None
    confidence: float = 0.0
    match_method: str = ""


class SourceProductSubmission(BaseModel):
    source_reference: str
    competitors: list[CompetitorMatch] = []

File: webapp/frontend/src/types/product.ts
export interface SourceProduct {
  reference: string
  name: string
  brand?: string
  ean?: string
  category?: string
  price?: number
  attributes?: Record<string, unknown>
}

export interface TargetProduct {
  reference: string
  name: string
  brand?: string
  ean?: string
  retailer?: string
  url?: string
  price?: number
  category?: string
  attributes?: Record<string, unknown>
}

export interface CompetitorMatch {
  reference: string
  competitor_retailer?: string
  competitor_product_name?: string
  competitor_url?: string
  competitor_price?: number
  confidence: number
  match_method: string
}

export interface SourceProductSubmission {
  source_reference: string
  competitors: CompetitorMatch[]
}

export interface MatchResult {
  category: string
  total_sources: number
  total_matches: number
  submissions: SourceProductSubmission[]
}
